lib/logthrsource/tests/test_logthrfetcherdrv.c

```diff
@@ -243,3 +243,114 @@ Test(logthrfetcherdrv, test_reconnect)
 
   destroy_test_threaded_fetcher(s);
 }
+
+static LogThreadedFetchResult
+_fetch_for_try_again_test(LogThreadedFetcherDriver *s)
+{
+  TestThreadedFetcherDriver *self = (TestThreadedFetcherDriver *) s;
+
+  if (__once())
```

Comment:
Fine for me. Do you have suggestion about the resolution? Should I copy paste the code, or should I promote to public? Can this remain in "messages.h" just rename from `__once` to `once`, or should I create a new place/header(like syslog-macros.h)?

Reply:
I'd use as local state as possible: `TestThreadedFetcherDriver`.

I don't think `once()` should be promoted or used anywhere. If you really want to use it in tests, it can be added to libtest, but I think it won't make us write good (unit) tests. `once` runs once per process but every location is a new separate context.

Reply:
I eliminated `once` by adding a state to `TestThreadedFetcherDriver`.